File: music-dlx-server/models/album.py
from dataclasses import dataclass

from .artist import Artist, ArtistResult


@dataclass
class Album:
    id: int
    title: str
    release_date: str
    year: int
    genres: list[str]
    artist: Artist
    cover: str


@dataclass
class AlbumResult:
    id: int
    title: str
    artist: ArtistResult
    cover: str

File: music-dlx-web/src/models/album.ts
import { IArtist, IArtistResult } from "./artist";

export interface IAlbum {
  id: number;
  title: string;
  release_date: string;
  year: number;
  genres: string[];
  artist: IArtist;
  cover: string;
}

export interface IAlbumResult {
  id: number;
  title: string;
  artist: IArtistResult;
  cover: string;
}
